lib/canonize.js

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+import {crypto} from './platform.js';
+import {encodeBase64Url} from './helpers.js';
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+// input: bytes from PDF417. returns base64Url encoded hash of canonized bytes.
+export async function base64UrlFromPdfBytes({pdfBytes} = {}) {
+  const hashedBytes = await hashFromPdfBytes({pdfBytes});
+  const base64Bytes = encodeBase64Url(hashedBytes);
+  return base64Bytes;
+}
+
+// input: bytes from Pdf. Returns CANONIZED BYTES
+// for privacy reasons, use hashCanonizedBytesFromPdfBytes
+// or base64UrlFromPdf417Bytes if sending data anywhere.
+export function canonizedBytesFromPdfBytes({pdfBytes} = {}) {
+
+  const pdfDataAsString = decoder.decode(pdfBytes);
+
+  // process PDF data
+  // assumption for parsing: DL is first subfile, 'DL' string first 2
+  // appearances are subfile indicator and subfile beginning
+  const subfile_indicator_index = pdfDataAsString.indexOf('DL');
+  const subfiles = pdfDataAsString.slice(pdfDataAsString.indexOf('DL',
+    subfile_indicator_index + 1) + 2);
```

Comment:
We perhaps haven't gotten the linter to do this yet (or we don't have the right linter config in this particular module), but our style is: if you drop any function param or array element, you drop them all:

```suggestion
  const subfiles = pdfDataAsString.slice(
    pdfDataAsString.indexOf('DL', subfile_indicator_index + 1) + 2);
```